### PR TITLE
vscode-extensions.b4dm4n.nixpkgs-fmt init at 0.0.1

### PR DESCRIPTION
**Pull request by zeratax**
###### Motivation for this change

add vscode extensions

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
do you mind rebasing on latest master instead of merging? `git pull -r origin master`

File: pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  mktplcRef = {
+    name = "nixpkgs-fmt";
+    publisher = "B4dM4n";
+    version = "0.0.1";
+    sha256 = "bf3da4537e81d7190b722d90c0ba65fd204485f49696d203275afb4a8ac772bf";
+  };
+
+  nativeBuildInputs = [
+    jq
+  ];
```
Comment (jonringer) [inline pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix:17]: is this just needed at build time? It won't be available when the extension is running

File: pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  preInstall = ''
+    jq '.contributes.configuration.properties."nixpkgs-fmt.path".default = $s' \
+      --arg s "${nixpkgs-fmt}/bin/nixpkgs-fmt" \
+      package.json >package.json.new
+    mv package.json.new package.json
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Format nix files with nixpkgs-fmt";
```
Comment (jonringer) [inline pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix:27]: please add homepage

File: pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix (hunk 1 above)
Comment (ZerataX) [reply pkgs/misc/vscode-extensions/nixpkgs-fmt/default.nix:17]: Yeah it's only needed to set the extension settings
rust-analyzer does something very similar:
https://github.com/NixOS/nixpkgs/blob/a37689ffd0490e3409948b8fe48f36f5c2e43044/pkgs/misc/vscode-extensions/rust-analyzer/default.nix#L33

**Comment by ZerataX** [comment]
I'm currently trying to rebase, but it goes through +10k commits with tons of merge issues, so i've decided to try
`git pull -r origin master -Xours` 
but I'm not sure if that will work. If not I might just create a new PR to supersede this one?

**Comment by ZerataX** [comment]
oh no sorry this is awful

**Pull request closed without merging**